src/SprykerEco/Zed/AmazonQuicksight/Business/Validator/QuicksightAnalyticsRequestValidator.php

```diff
@@ -0,0 +1,167 @@
+     *
+     * @return bool
+     */
+    public function isResetAnalyticsEnabled(
+        ?QuicksightAssetBundleImportJobTransfer $quicksightAssetBundleImportJobTransfer,
+        ?QuicksightUserTransfer $quicksightUserTransfer
+    ): bool {
+        return $this->isAssetBundleSuccessfullyInitialized($quicksightAssetBundleImportJobTransfer)
+            && !$this->isAssetBundleInitializationInProgress($quicksightAssetBundleImportJobTransfer)
+            && $this->isQuicksightUserAuthor($quicksightUserTransfer);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuicksightAssetBundleImportJobTransfer|null $quicksightAssetBundleImportJobTransfer
+     *
+     * @return bool
+     */
+    public function isEnableAnalyticsEnabled(
+        ?QuicksightAssetBundleImportJobTransfer $quicksightAssetBundleImportJobTransfer
+    ): bool {
+        return !$this->isAssetBundleSuccessfullyInitialized($quicksightAssetBundleImportJobTransfer)
+            && !$this->isAssetBundleInitializationInProgress($quicksightAssetBundleImportJobTransfer);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuicksightAssetBundleImportJobTransfer|null $quicksightAssetBundleImportJobTransfer
+     *
+     * @return bool
+     */
+    public function isAssetBundleSuccessfullyInitialized(
+        ?QuicksightAssetBundleImportJobTransfer $quicksightAssetBundleImportJobTransfer
+    ): bool {
+        return $quicksightAssetBundleImportJobTransfer
+            && $quicksightAssetBundleImportJobTransfer->getIsInitialized();
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuicksightAssetBundleImportJobTransfer|null $quicksightAssetBundleImportJobTransfer
+     *
+     * @return bool
+     */
+    public function isAssetBundleInitializationInProgress(
+        ?QuicksightAssetBundleImportJobTransfer $quicksightAssetBundleImportJobTransfer
+    ): bool {
+        return $quicksightAssetBundleImportJobTransfer
+            && !in_array($quicksightAssetBundleImportJobTransfer->getStatus(), $this->amazonQuicksightConfig->getAssetBundleImportJobCompletionStatuses());
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuicksightUserTransfer|null $quicksightUserTransfer
+     *
+     * @return bool
+     */
+    public function isQuicksightUserRoleAvailable(?QuicksightUserTransfer $quicksightUserTransfer): bool
+    {
+        return $quicksightUserTransfer && in_array(
+            $quicksightUserTransfer->getRole(),
+            $this->amazonQuicksightConfig->getQuicksightUserRoles(),
+        );
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuicksightUserTransfer|null $quicksightUserTransfer
+     *
+     * @return bool
+     */
+    protected function isQuicksightUserAuthor(?QuicksightUserTransfer $quicksightUserTransfer): bool
+    {
+        return $quicksightUserTransfer && $quicksightUserTransfer->getRole() === static::QUICKSIGHT_USER_ROLE_AUTHOR;
```

Comment:
[minor] Same here, for extendability we can create a const with author roles because quicksight has two roles: author and author pro